### PR TITLE
Fix Clang warnings

### PR DESCRIPTION
**Pull request by staticssleever668**

### CODE REVIEW

**Comment by staticssleever668** [comment]
@Gegel85.

**Comment by PinkySmile** [comment]
I think it needs %zu for all strlen calls too because it returns a size_t too and on windows it is either unsigned int in 32bits or unsigned long long in 64bits.

**Comment by staticssleever668** [comment]
@Gegel85, yeah, you are right. This time I also went ahead and fixed clang-tidy warnings (the only warning left is about `strerror` function being thread-unsafe).

File: src/osu_map_parser.c [hunk 1]
```diff
@@ -28,13 +28,13 @@ size_t	OsuMap_getStringArraySize(char **elems)
...
-	void	*buffer;
-	char	**array;
+	void	*buffer = NULL;
+	char	**array = NULL;
 	size_t	size = 1;
 
 	array = malloc(2 * sizeof(*array));
 	if (!array){
-		sprintf(error_buffer, "Memory allocation error (%luB)", (unsigned long)(2 * sizeof(*array)));
+		sprintf(error_buffer, "Memory allocation error (%luB)", 2 * sizeof(*array));
```
Comment (PinkySmile) [inline src/osu_map_parser.c]: That is supposed to be %zu

File: src/osu_map_parser.c [hunk 2]
```diff
@@ -164,7 +164,7 @@ OsuMapCategory	*OsuMap_getCategories(char **lines, char *error_buffer, jmp_buf j
 	if (!lines[0]) {
 		categories = calloc(1, sizeof(*categories));
 		if (!categories) {
-			sprintf(error_buffer, "Memory allocation error (%luB)", (unsigned long)sizeof(*categories));
+			sprintf(error_buffer, "Memory allocation error (%luB)", sizeof(*categories));
```
Comment (PinkySmile) [inline src/osu_map_parser.c]: I think that too

File: src/osu_map_parser.c [hunk 3]
```diff
@@ -192,7 +192,7 @@ OsuMapCategory	*OsuMap_getCategories(char **lines, char *error_buffer, jmp_buf j
 		for (; lines[i] && lines[i][0] != '['; i++);
 		categories[currentIndex].lines = malloc((i - start + 1) * sizeof(*categories[currentIndex].lines));
 		if (!categories[currentIndex].lines) {
-			sprintf(error_buffer, "Memory allocation error (%luB)", (unsigned long)((i - start + 1) * sizeof(*categories[currentIndex].lines)));
+			sprintf(error_buffer, "Memory allocation error (%luB)", (i - start + 1) * sizeof(*categories[currentIndex].lines));
```
Comment (PinkySmile) [inline src/osu_map_parser.c]: Here

File: src/osu_map_parser.c [hunk 4]
```diff
@@ -386,7 +382,7 @@ OsuMap_unsignedIntegerArray	OsuMap_getCategoryElementUIntegerArray(char **lines,
 				result.length += lines[i][j] == ',';
 			result.content = malloc((result.length + 1) * sizeof(*result.content));
 			if (!result.content) {
-				sprintf(err_buffer, "Memory allocation error (%luB)", (unsigned long)(result.length * sizeof(*result.content)));
+				sprintf(err_buffer, "Memory allocation error (%luB)", result.length * sizeof(*result.content));
```
Comment (PinkySmile) [inline src/osu_map_parser.c]: zu

File: src/osu_map_parser.c [hunk 5]
```diff
@@ -561,8 +557,8 @@ OsuIntegerVector	OsuMap_getIntegerVector(char *str, char *err_buffer, jmp_buf ju
 		free(elems);
 		longjmp(jump_buffer, true);
 	}
-	vector.x = OsuMap_getInteger(elems[0], 0, 0, err_buffer, jump_buffer);
-	vector.y = OsuMap_getInteger(elems[1], 0, 0, err_buffer, jump_buffer);
+	vector.x = (int)OsuMap_getInteger(elems[0], 0, 0, err_buffer, jump_buffer);
+	vector.y = (int)OsuMap_getInteger(elems[1], 0, 0, err_buffer, jump_buffer);
```
Comment (PinkySmile) [inline src/osu_map_parser.c]: I think the cleaner fix would be to change the type for the vectors to be longs instead

File: src/osu_map_parser.c [hunk 6]
```diff
@@ -576,7 +572,7 @@ OsuIntegerVectorArray	OsuMap_getIntegerVectorArray(char *str, char *err_buffer,
 	array.length = OsuMap_getStringArraySize(elems);
 	array.content = malloc(array.length * sizeof(*array.content));
 	if (!array.content) {
-		sprintf(err_buffer, "Memory allocation error (%luB)", (unsigned long)(array.length * sizeof(*array.content)));
+		sprintf(err_buffer, "Memory allocation error (%luB)", array.length * sizeof(*array.content));
```
Comment (PinkySmile) [inline src/osu_map_parser.c]: zu too probably

File: src/osu_map_parser.c [hunk 7]
```diff
@@ -643,8 +639,7 @@ OsuMap_hitObjectSliderInfos	OsuMap_getSliderInfos(char **elems, char *err_buffer
 		}
 		infos.edgeHitsounds = malloc(OsuMap_getStringArraySize(nbr) + 1);
 		if (!infos.edgeHitsounds) {
-			sprintf(err_buffer, "Memory allocation error (%luB)",
-				(unsigned long)(OsuMap_getStringArraySize(nbr) + 1));
+			sprintf(err_buffer, "Memory allocation error (%luB)", OsuMap_getStringArraySize(nbr) + 1);
```
Comment (PinkySmile) [inline src/osu_map_parser.c]: zu

File: src/osu_map_parser.c [hunk 8]
```diff
@@ -662,15 +657,14 @@ OsuMap_hitObjectSliderInfos	OsuMap_getSliderInfos(char **elems, char *err_buffer
 			}
 			infos.edgeAdditions = malloc((OsuMap_getStringArraySize(nbr) + 1) * sizeof(*infos.edgeAdditions));
 			if (!infos.edgeAdditions) {
-				sprintf(err_buffer, "Memory allocation error (%luB)",
-					(unsigned long)((OsuMap_getStringArraySize(nbr) + 1) * sizeof(*infos.edgeAdditions)));
+				sprintf(err_buffer, "Memory allocation error (%luB)", (OsuMap_getStringArraySize(nbr) + 1) * sizeof(*infos.edgeAdditions));
```
Comment (PinkySmile) [inline src/osu_map_parser.c]: zu

File: src/osu_map_parser.c [hunk 9]
```diff
@@ -688,27 +682,27 @@ OsuMap_hitObject	OsuMap_parseLineToHitObject(char *line, char *err_buffer, jmp_b
...
-			sprintf(err_buffer, "Invalid hit object infos '%s': 8-11 fields expected but %i found", old, len);
+			sprintf(err_buffer, "Invalid hit object infos '%s': 8-11 fields expected but %zu found", old, len);
 			free(elems);
 			free(old);
 			longjmp(jump_buffer, true);
 		}
 		obj.additionalInfos = malloc(sizeof(OsuMap_hitObjectSliderInfos));
 		if (!obj.additionalInfos) {
-			sprintf(err_buffer, "Memory allocation error (%luB)", (unsigned long)sizeof(OsuMap_hitObjectSliderInfos));
+			sprintf(err_buffer, "Memory allocation error (%luB)", sizeof(OsuMap_hitObjectSliderInfos));
```
Comment (PinkySmile) [inline src/osu_map_parser.c]: zu

File: src/osu_map_parser.c [hunk 10]
```diff
@@ -719,14 +713,14 @@ OsuMap_hitObject	OsuMap_parseLineToHitObject(char *line, char *err_buffer, jmp_b
...
-			sprintf(err_buffer, "Invalid hit object infos '%s': 7 fields expected but %i found", old, len);
+			sprintf(err_buffer, "Invalid hit object infos '%s': 7 fields expected but %zu found", old, len);
 			free(elems);
 			free(old);
 			longjmp(jump_buffer, true);
 		}
 		obj.additionalInfos = malloc(sizeof(unsigned long));
 		if (!obj.additionalInfos) {
-			sprintf(err_buffer, "Memory allocation error (%luB)", (unsigned long)sizeof(unsigned long));
+			sprintf(err_buffer, "Memory allocation error (%luB)", sizeof(unsigned long));
```
Comment (PinkySmile) [inline src/osu_map_parser.c]: zu

File: src/osu_map_parser.c [hunk 11]
```diff
@@ -763,7 +757,7 @@ OsuMap_hitObjectArray	OsuMap_getCategoryHitObject(OsuMapCategory *category, char
 	for (; category->lines[elements.length]; elements.length++);
 	elements.content = malloc(elements.length * sizeof(*elements.content));
 	if (!elements.content) {
-		sprintf(err_buffer, "Memory allocation error (%luB)", (unsigned long)(elements.length * sizeof(*elements.content)));
+		sprintf(err_buffer, "Memory allocation error (%luB)", elements.length * sizeof(*elements.content));
```
Comment (PinkySmile) [inline src/osu_map_parser.c]: zu

File: src/osu_map_parser.c [hunk 12]
```diff
@@ -831,7 +825,7 @@ OsuMap_colorArray	OsuMap_getCategoryColors(OsuMapCategory *category, char *err_b
 		elements.length += OsuMap_stringStartsWith(category->lines[i], "Combo");
 	elements.content = malloc(elements.length * sizeof(*elements.content));
 	if (!elements.content) {
-		sprintf(err_buffer, "Memory allocation error (%luB)", (unsigned long)(elements.length * sizeof(*elements.content)));
+		sprintf(err_buffer, "Memory allocation error (%luB)", elements.length * sizeof(*elements.content));
```
Comment (PinkySmile) [inline src/osu_map_parser.c]: zu

File: src/osu_map_parser.c [hunk 13]
```diff
@@ -862,7 +856,7 @@ OsuMap_timingPointArray	OsuMap_getCatergoryTimingPoints(OsuMapCategory *category
 	for (; category->lines[elements.length]; elements.length++);
 	elements.content = malloc(elements.length * sizeof(*elements.content));
 	if (!elements.content) {
-		sprintf(err_buffer, "Memory allocation error (%luB)", (unsigned long)(elements.length * sizeof(*elements.content)));
+		sprintf(err_buffer, "Memory allocation error (%luB)", elements.length * sizeof(*elements.content));
```
Comment (PinkySmile) [inline src/osu_map_parser.c]: zu

**Review by PinkySmile — CHANGES_REQUESTED** [review]
I marked all the spots that probably should be zu too (though I'm not sure)

**Comment by PinkySmile** [comment]
Btw, how did you get here ? x)

**Comment by staticssleever668** [comment]
@Gegel85, hi, thanks for the review! I fixed those as well.

I came here by searching osu! parsers in GitHub. I needed one in C/C++ and didn't want to write it by myself at the time, and this one looked the most promising.

**Review by PinkySmile — COMMENTED** [review]
Everything seems fine. I'll just pull the submodule in my replay player just to check if it didn't break anything relevent before merging.

File: include/osu_map_parser.h [hunk 14]
```diff
@@ -150,20 +150,20 @@ typedef struct OsuMap_hitObjectAddition {
 
 typedef struct OsuIntegerVectorArray {
```
Comment (PinkySmile) [inline include/osu_map_parser.h]: You forgot to change it here x)

**Review by PinkySmile — CHANGES_REQUESTED** [review]
You just forgot to rename a struct. Besides that, everything seems fine and ready for merge.

File: include/osu_map_parser.h (hunk 14 above)
Comment (staticssleever668) [reply include/osu_map_parser.h]: Fixed.

**Review by PinkySmile — APPROVED** [review]

**Comment by PinkySmile** [comment]
All good ! Thanks for helping !

**Pull request merged**